apps/fz_http/lib/fz_http_web/controllers/auth_controller.ex

```diff
@@ -48,6 +49,29 @@ defmodule FzHttpWeb.AuthController do
     end
   end
 
+  def callback(conn, params) do
+    %{"provider" => provider} = params
+
+    with {:ok, tokens} <- OpenIDConnect.fetch_tokens(provider, params),
+         {:ok, claims} <- OpenIDConnect.verify(provider, tokens["id_token"]) do
+      case UserFromAuth.find_or_create(provider, claims) do
+        {:ok, user} ->
+          conn
+          |> configure_session(renew: true)
+          |> put_session(:live_socket_id, "users_socket:#{user.id}")
+          |> Authentication.sign_in(user, %{provider: provider})
+          |> redirect(to: root_path_for_role(conn, user.role))
+
+        {:error, reason} ->
+          conn
+          |> put_flash(:error, "Error signing in: #{reason}")
+          |> request(%{})
+      end
+    else
+      _ -> send_resp(conn, 401, "")
```

Comment:
@mdp Would it make sense to log the error message here with something like (apologies have not tested this):

```
{:error, reason} ->
  Logger.warn("OpenIDConnect Error: Could not fetch tokens from params. Reason: #{reason}")
  send_resp(conn, 401, "")
```

Just so the administrator has something to debug if this fails?

Reply:
Yeah, I think that makes sense.